src/plugins/intel_gpu/include/intel_gpu/runtime/memory.hpp

```diff
@@ -225,6 +225,7 @@ struct surfaces_lock {
     surfaces_lock& operator=(const surfaces_lock& other) = delete;
 
     static std::unique_ptr<surfaces_lock> create(engine_types engine_type, std::vector<memory::ptr> mem, const stream& stream);
+    static bool is_surface_lock_check_needed(const shared_mem_type& mem_type);
 };
```

Comment:
random spot) did you review to make a test case for this behavior? is it difficult?

Reply:
After reviewed for test case, but it's not easy to make the case because error is not deterministic so we can't reproduce it in simple test case.